pkg/api/index.go

```diff
@@ -152,7 +154,42 @@ func (hs *HTTPServer) getAppLinks(c *models.ReqContext) ([]*dtos.NavLink, error)
 			if len(appLink.Children) == 1 && appLink.Children[0].Url == appLink.Url {
 				appLink.Children = []*dtos.NavLink{}
 			}
-			appLinks = append(appLinks, appLink)
+
+			if navId, hasNavId := hs.Cfg.PluginAppNavIds[plugin.ID]; hasNavId && topNavEnabled {
+				if navNode := navlinks.FindById(navTree, navId); navNode != nil {
+					navNode.Children = append(navNode.Children, appLink)
+				} else {
+					// Lazily add this node
+					if navId == "monitoring" {
+						navTree = append(navTree, &dtos.NavLink{
+							Text:        "Monitoring",
+							Id:          "monitoring",
+							Description: "Monitoring and infrastructure apps",
+							Icon:        "monitor",
+							Section:     dtos.NavSectionCore,
+							Children:    []*dtos.NavLink{appLink},
+							Url:         hs.Cfg.AppSubURL + "/monitoring",
+						})
+					}
+					if navId == "alerts-and-incidents" {
+						alertingNode := navlinks.FindById(navTree, "alerting")
+						navTree = append(navTree, &dtos.NavLink{
+							Text:        "Alerts & incidents",
+							Id:          "alerts-and-incidents",
+							Description: "Alerting and incident management apps",
+							Icon:        "bell",
+							Section:     dtos.NavSectionCore,
+							Children:    []*dtos.NavLink{alertingNode, appLink},
+							Url:         hs.Cfg.AppSubURL + "/alerts-and-incidents",
+						})
+						// Remove alerting node from navTree roots
+						navTree = navlinks.FilterOutById(navTree, "alerting")
+					}
+					hs.log.Error("Plugin app nav id not found", "pluginId", plugin.ID, "navId", navId)
+				}
+			} else {
+				appLinks = append(appLinks, appLink)
+			}
```

Comment:
Would it be possible to introduce some `continue` statements to make early exits and prevent nesting? Also, if you define a `navId` that doesn't exist. Should we add the app to the regular "apps" section? Or do we want it to be hidden so you need to correct your config?

```suggestion
			if navId, hasNavId := hs.Cfg.PluginAppNavIds[plugin.ID]; hasNavId && topNavEnabled {
				if navNode := navlinks.FindById(navTree, navId); navNode != nil {
					navNode.Children = append(navNode.Children, appLink)
					continue
				}
				// Lazily add this node

				if navId == "monitoring" {
					navTree = append(navTree, &dtos.NavLink{
						Text:        "Monitoring",
						Id:          "monitoring",
						Description: "Monitoring and infrastructure apps",
						Icon:        "monitor",
						Section:     dtos.NavSectionCore,
						Children:    []*dtos.NavLink{appLink},
						Url:         hs.Cfg.AppSubURL + "/monitoring",
					})
					continue
				}

				if navId == "alerts-and-incidents" {
					alertingNode := navlinks.FindById(navTree, "alerting")
					navTree = append(navTree, &dtos.NavLink{
						Text:        "Alerts & incidents",
						Id:          "alerts-and-incidents",
						Description: "Alerting and incident management apps",
						Icon:        "bell",
						Section:     dtos.NavSectionCore,
						Children:    []*dtos.NavLink{alertingNode, appLink},
						Url:         hs.Cfg.AppSubURL + "/alerts-and-incidents",
					})
					// Remove alerting node from navTree roots
					navTree = navlinks.FilterOutById(navTree, "alerting")
					continue
				}

				hs.log.Error("Plugin app nav id not found", "pluginId", plugin.ID, "navId", navId)
				// continue - if we don't want the app to fallback to the existing app section.
			}

			appLinks = append(appLinks, appLink)
			continue
		}
```